app/components/newsletter/newsletter-header-overview.js

```diff
@@ -106,73 +72,151 @@ export default class NewsletterHeaderOverviewComponent extends Component {
 
     await reloadedMeeting.save();
     this.isLoading = false;
-    location.reload();
   }
 
   @action
-  async sendCampaign() {
+  async publishToMail() {
     this.isLoading = true;
-    const agenda = this.args.agenda;
-    const meeting = this.args.meeting;
-    const mailCampaign = await meeting.mailCampaign;
+    const mailCampaign = await this.getMailCampaign();
 
-    if (!mailCampaign || !mailCampaign.id || mailCampaign.isSent) {
-      return;
+    if (mailCampaign) {
+      if (await this.validateMailCampaign(mailCampaign)) {
+        await this.publishToMailAndSaveCampaign(mailCampaign);
+      }
+    }
+
+    this.isLoading = false;
+    this.toggleVerifyingPublishMail();
+  }
+
+  @action
+  async publishToBelga() {
+    this.isLoading = true;
+    await this.publishNewsletterToBelga();
+    this.isLoading = false;
+    this.toggleVerifyingPublishBelga();
+  }
+
+  @action
+  async publishToAll() {
+    this.isLoading = true;
+
+    const mailCampaign = await this.getMailCampaign();
+    if (mailCampaign) {
+      if (await this.validateMailCampaign(mailCampaign)) {
+        await this.publishToMailAndSaveCampaign(mailCampaign);
+      }
+    }
+    await this.publishNewsletterToBelga;
+
+    this.isLoading = false;
+    this.toggleVerifyingPublishAll();
+  }
+
+  @action
+  async showNewsletter() {
+    this.loadingNewsletter = true;
+
+    const mailCampaign = await this.getMailCampaign();
+
+    if (mailCampaign) {
+      const html = await this.newsletterService
+        .getMailCampaignContent(mailCampaign.campaignId)
+        .catch(() => {
+          this.toaster.error(
+            this.intl.t('error-send-newsletter'),
+            this.intl.t('warning-title')
+          );
+        });
+      this.newsletterHTML = html.body;
     }
+    this.loadingNewsletter = false;
+  }
+
+  @action
+  async clearNewsletterHTML() {
+    this.newsletterHTML = null;
+    this.loadingNewsletter = false;
+  }
 
+  async validateMailCampaign(mailCampaign) {
     let validCampaign = true;
-    await this.validatedCampaign(mailCampaign.campaignId).catch((ex) => {
-      console.error(ex);
-      this.isVerifying = false;
-      this.isLoading = false;
+
+    const campaign = await this.newsletterService
+      .getMailCampaign(mailCampaign.campaignId)
+      .catch(() => {
+        this.toaster.error(
+          this.intl.t('error-fetch-newsletter'),
+          this.intl.t('warning-title')
+        );
+        validCampaign = false;
+      });
```

Comment:
Deze manier van `catch` en `error` kunnen we beter afhandelen in een try catch.
Wel wat buiten scope, mogelijk past dat beter in de "json api compliance" story

Reply:
Ik neem het mee